go.mod

```diff
@@ -5,18 +5,28 @@ go 1.13
 require (
 	cloud.google.com/go v0.37.4 // indirect
 	github.com/coreos/prometheus-operator v0.34.0
+	github.com/go-logr/logr v0.1.0
```

Comment:
This package addition is bringing in additional indirect dependencies and a lot of new packages under the vendor folder. Is there a reason to use this over `"sigs.k8s.io/controller-runtime/pkg/runtime/log"`?

https://github.com/integr8ly/heimdall/blob/master/pkg/controller/deployments/controller.go#L25

Reply:
The only reason this package was added was to declare the `logr.Logger` type as a member of the `Reconciler` struct: https://github.com/integr8ly/heimdall/pull/32/files#diff-1acebcd3c7162bca673972c513212b1dR52

Variables of this type are being used already but the type was never explicitely declared so I guess that's why it wasn't added to the `go.mod` until now. Not sure if there's any way around it

Reply:
Thanks @sergioifg94 . I am not seeing where the `logr` package was being used directly already. I think that is what the controller-runtime log package might use under the hood. My preference would be to stay consistent with the existing controllers here and use the controller-runtime log package the same as https://github.com/integr8ly/heimdall/blob/master/pkg/controller/deployments/controller.go#L25

Reply:
@davidffrench That's the function where I got the `logr.Logger` type. It returns an instance of this type but, because so far it has been used with type inference, the type was never explicitely declared. A workaround for this is to let the generic controller create the logger and receive the name so we can avoid importing the `logr` package and keep using it

Reply:
I think it is fine to allow the type inference. You can continue using the controller-runtime abstraction on logr as is used in other controllers. The package is still included in our vendor folder as it is a dependency of `controller-runtime` log package. https://github.com/integr8ly/heimdall/tree/master/vendor/github.com/go-logr

Reply:
@davidffrench Just updated the PR. Had to create an interface with the same methods and fortunately go's type system is smart enough to validate that `logr.Logger` is assignable to that interface even though they have a different name, so I don't have to explicitly import the package